xrtmodels/response.go

```diff
@@ -2,33 +2,51 @@
 
 package xrtmodels
 
-import "github.com/edgexfoundry/go-mod-core-contracts/v2/v1models"
+import (
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/errors"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+)
+
+const (
+	// https://github.com/IOTechSystems/xrt/blob/master/include/devsdk/devsdk.h#L27
+	XrtSdkStatusOk               = 0
+	XrtSdkStatusNotFound         = 1
+	XrtSdkStatusNotSupported     = 2
+	XrtSdkStatusInvalidOperation = 3
+	XrtSdkStatusAlreadyExists    = 7
+)
 
 type BaseResponse struct {
 	Client    string `json:"client"`
 	RequestId string `json:"request_id"`
+	Type      string `json:"type"`
 }
 
 type CommonResponse struct {
 	BaseResponse `json:",inline"`
 	Result       BaseResult `json:"result"`
-	Error        string     `json:"error"`
-}
-
-// ErrorMessage return the error message from the XRT response
-// The response format might be {"client":"...","error":"...","request_id":"...","result":{"success":false},"success":false}
-// or {"client":"...","request_id":"...","result":{"error":"...","success":false}}
-func (c CommonResponse) ErrorMessage() string {
-	if c.Error != "" {
-		return c.Error
-	} else {
-		return c.Result.Error
-	}
 }
 
 type BaseResult struct {
-	Success bool   `json:"success"`
-	Error   string `json:"error"`
+	Status       int    `json:"status"`
+	ErrorMessage string `json:"error_message"`
+}
+
+func (result BaseResult) Error() errors.EdgeX {
+	switch result.Status {
+	case XrtSdkStatusOk:
+		return nil
+	case XrtSdkStatusNotFound:
+		return errors.NewCommonEdgeX(errors.KindEntityDoesNotExist, result.ErrorMessage, nil)
+	case XrtSdkStatusNotSupported:
+		return errors.NewCommonEdgeX(errors.KindServiceUnavailable, result.ErrorMessage, nil)
```

Comment:
should it be `KindNotImplemented`?

Reply:
The XRT will throw the `DEVSDK_STATUS_NOT_SUPPORTED` status code when `discovery not supported`, so I think it's fine to use `KindNotImplemented`.
https://github.com/IOTechSystems/xrt/blob/v2.0-branch/src/c/devsdk/operations.c

Reply:
In this case, KindNotImplemented fit the situation

Reply:
Got it, modified.